### PR TITLE
Use database services in ListItems channel provider.

### PR DESCRIPTION
**Pull request by demiankatz**
This PR adjusts the ListItems channel provider to use database services instead of direct table access. I believe it makes the code significantly clearer, though it may in some situations be less efficient than the previous version. I'm not overly concerned about efficiency because I doubt that this is being used in particularly complex ways in the wild, and we can improve it in future if the need arises. (I think some other enhancements, like setting limits on how many lists can be retrieved, would probably be wise as well... but that's out of scope for the present refactoring).

### CODE REVIEW

File: module/VuFind/src/VuFind/Db/Service/UserListService.php [hunk 1]
```diff
@@ -96,6 +96,33 @@ public function getUserListById(int $id): UserListEntityInterface
         return $result;
     }
 
+    /**
+     * Get public lists.
+     *
+     * @param array $includeFilter List of list ids or entities to include in result.
+     * @param array $excludeFilter List of list ids or entities to exclude from result.
+     *
+     * @return UserEntityInterface[]
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Service/UserListService.php]: `UserListEntityInterface[]`?

File: module/VuFind/src/VuFind/Db/Service/UserListServiceInterface.php [hunk 2]
```diff
@@ -72,6 +72,16 @@ public function deleteUserList(UserListEntityInterface|int $listOrId): void;
      */
     public function getUserListById(int $id): UserListEntityInterface;
 
+    /**
+     * Get public lists.
+     *
+     * @param array $includeFilter List of list ids or entities to include in result.
+     * @param array $excludeFilter List of list ids or entities to exclude from result.
+     *
+     * @return UserEntityInterface[]
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Service/UserListServiceInterface.php]: `UserListEntityInterface`?

File: module/VuFind/src/VuFind/Db/Service/UserListService.php [hunk 3]
```diff
@@ -142,6 +169,33 @@ public function getUserListsAndCountsByUser(UserEntityInterface|int $userOrId):
...
+     * @return UserListEntityInterface[]
+     */
+    public function getUserListsByTagAndId(
+        string|array|null $tag = null,
+        int|array|null $listId = null,
+        bool $publicOnly = true,
+        bool $andTags = true
+    ): array {
+        $listIds = array_column(
+            iterator_to_array($this->getDbTable('ResourceTags')->getListsForTag($tag, $listId, $publicOnly, $andTags)),
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Service/UserListService.php:190]: `getUserListsByTagAndId()` allows `$tag` to be `null`, but `getListsForTag()` does not?

**Review by aleksip — APPROVED** [review]

File: module/VuFind/src/VuFind/Db/Service/UserListService.php (hunk 3 above)
Comment (demiankatz) [reply module/VuFind/src/VuFind/Db/Service/UserListService.php:190]: Good catch; I think it was working in spite of this, but I've added the support more explicitly in commit 0d723bfd870c0b7d780eb3976a1b591f21aa0fbd.

**Pull request merged**